Framework/Algorithms/src/GetEi.cpp

```diff
@@ -367,22 +366,26 @@ void GetEi::extractSpec(int wsInd, double start, double end) {
 */
 void GetEi::getPeakEstimates(double &height, int64_t &centreInd,
                              double &background) const {
+
+  const auto &X = m_tempWS->x(0);
+  const auto &Y = m_tempWS->y(0);
   // take note of the number of background counts as error checking, do we have
   // a peak or just a bump in the background
   background = 0;
   // start at the first Y value
-  height = m_tempWS->readY(0)[0];
+  height = Y[0];
   centreInd = 0;
+
   // then loop through all the Y values and find the tallest peak
-  for (MantidVec::size_type i = 1; i < m_tempWS->readY(0).size() - 1; ++i) {
-    background += m_tempWS->readY(0)[i];
-    if (m_tempWS->readY(0)[i] > height) {
+  for (size_t i = 1; i < Y.size() - 1; ++i) {
+    background += Y[i];
```

Comment:
How do I go on about using `std::max` because the for loop also saves the index of the max element?